src/EnergyPlus/api/runtime.cc

```diff
@@ -104,16 +104,23 @@ void registerProgressCallback(EnergyPlusState EP_UNUSED(state), void (*f)(int co
     EnergyPlus::DataGlobals::progressCallback = f;
 }
 
-void registerStdOutCallback(EnergyPlusState EP_UNUSED(state), void (*f)(const char * message)) {
+void registerStdOutCallback(EnergyPlusState EP_UNUSED(state), std::function<void (const std::string &)> f) {
     EnergyPlus::DataGlobals::messageCallback = f;
 }
 
+void registerStdOutCallback(EnergyPlusState state, void (*f)(const char * message)) {
+    const auto stdf = [f](const std::string & message) {
+        f(message.c_str());
+    };
+    registerStdOutCallback(state, std::function<void (const std::string &)>(stdf));
+}
```

Comment:
I feel like really we should invert all the register callbacks to this approach.